kong/clustering/data_plane.lua

```diff
@@ -80,10 +80,50 @@ function _M:init_worker(basic_info)
   self.plugins_list = basic_info.plugins
   self.filters = basic_info.filters
 
-  -- only run in process which worker_id() == 0
-  assert(ngx.timer.at(0, function(premature)
-    self:communicate(premature)
-  end))
+  local function start_communicate()
+    assert(ngx.timer.at(0, function(premature)
+      self:communicate(premature)
+    end))
+  end
+
+  -- does not config rpc sync
+  if not kong.sync then
+    start_communicate()
+    return
+  end
+
+  local worker_events = assert(kong.worker_events)
+
+  -- if rpc is ready we will check then decide how to sync
+  worker_events.register(function(capabilities_list)
+    -- we only check once
+    if self.inited then
```

Comment:
I suppose it's fine to make another sync call when reconnecting?

Reply:
Of cause.